src/components/browse/sidebar/searchbar.tsx

```diff
@@ -3,20 +3,36 @@ import { Form, Button, Container } from "react-bootstrap";
 import { Search } from "react-bootstrap-icons";
 import { searchResponseModel } from "../../../models/dataset";
 import { getDatasetsSearchResp } from "../../../api/browse";
+import { useNavigate } from 'react-router-dom'
 
 interface searchbarProps {
   setSearchResp: Dispatch<SetStateAction<searchResponseModel | null>>;
   setSearchKeyword: Dispatch<SetStateAction<string>>;
   searchKeyword: string;
   limit: number;
+  searchParams: any
+  setSearchParams: any
+  setPage: Dispatch<SetStateAction<number>>;
 }
 
 const Searchbar = (props: searchbarProps) => {
-
+  let navigate = useNavigate();
   const skip = 0;
   const handleSearch = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault()
     getDatasetsSearchResp(props.setSearchResp, [], props.searchKeyword, skip, props.limit);
+    if (props.searchParams.q === undefined) {
+      props.setSearchParams({ q: props.searchKeyword })
+      props.setSearchParams({ p: 1 })
```

Comment:
The p is set to be default to 0 in browse, why are we doing 1 were and in other areas?

Reply:
Changed behaviour to my suggestion. Resolved.